capsules/src/button.rs

```diff
@@ -81,18 +81,29 @@ pub enum ButtonState {
 /// Manages the list of GPIO pins that are connected to buttons and which apps
 /// are listening for interrupts from which buttons.
 pub struct Button<'a> {
-    pins: &'a [(&'a dyn gpio::InterruptValuePin, GpioMode)],
+    pins: &'a [(
+        &'a dyn gpio::InterruptValuePin,
+        GpioMode,
+        Option<gpio::FloatingState>,
+    )],
     apps: Grant<(Option<Callback>, SubscribeMap)>,
 }
 
 impl<'a> Button<'a> {
     pub fn new(
-        pins: &'a [(&'a dyn gpio::InterruptValuePin, GpioMode)],
+        pins: &'a [(
+            &'a dyn gpio::InterruptValuePin,
+            GpioMode,
+            Option<gpio::FloatingState>,
```

Comment:
I don't think we need `Some(gpio::FloatingState::PullNone)` and `None`, so I think we can just remove this option.

Reply:
Several boards are not defining the FloatingState for their buttons at the moment. It means that this will default to the on-reset value which is probably chip-specific.
Explictly putting `FloatingState::PullNone` will put the pin in HiZ if external resistors are not present (the exact case we faced on the Nordic boards). I don't know if it's better to active the internal resistors in all cases (the consequence will be non-optimal power consumption and power dissipation).

Reply:
Gotcha. I think if we just match what the datasheets say we should be fine. From my looking:

- Arty-e21: by default input pins are floating
- nucleo boards: by default input pins are floating

Reply:
Done.